pegasus/router.rb

```diff
@@ -479,6 +479,8 @@ def resolve_template(subdir, extnames, uri, is_document: false)
       # Also look for shared items.
       found = MultipleExtnameFileUtils.find_with_extnames(content_dir('..', '..', 'shared', 'haml'), uri, extnames)
       return found.first unless found.empty?
+    rescue Errno::ENAMETOOLONG
+      # Just return nothing when the URL is malformed by being too long
```

Comment:
Since we’re nearly certain that this error is being caused by a problem with the client request, maybe we should return an HTTP Response Status code in the 4XX range here?